src/modules/hip/kernel/down_mixing.hpp

```diff
@@ -0,0 +1,64 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+__global__ void down_mixing_hip_tensor(float *srcPtr,
+                                       uint srcStride,
+                                       float *dstPtr,
+                                       uint dstStride,
+                                       int *srcDimsTensor)
+
+{
+    int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+    int srcLength = srcDimsTensor[id_z * 2];
+    int channels = srcDimsTensor[id_z * 2 + 1];
+
+    if (id_x >= srcLength)
+        return;
+
+    // multi channel
+    if(channels > 1)
+    {
+        float nomalizedWeight = 1.f / channels;
+        float outVal = 0.0f;
+        uint srcIdx = id_z * srcStride + id_x * channels;
+        for(int i = 0; i < channels; i++, srcIdx++)
+            outVal += srcPtr[srcIdx] * nomalizedWeight;
+
+        uint dstIdx = id_z * dstStride + id_x;
+        dstPtr[dstIdx] = outVal;
+    }
+    // single channel - copy input to output
+    else
```

Comment:
I feel that the code inside the 'if' condition will work for a single channel as well. Can we remove the code in the 'else' part and have a generic code  that works for both single and multi-channel inputs?